Updates sort package to latest

sm43 opened the pull request:
# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

we use sort code from k8s repo, this updates the code to the latest master
https://github.com/kubernetes/kubernetes/blob/20d0ab7ae808aaddb1556c3c38ca0607663c50ac/staging/src/k8s.io/kubectl/pkg/cmd/get/sorter.go#L150
but keeps using integer.IntMin to continue using golang 1.20


<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] 📝 Please ensure your commit message is clear and informative. For guidance on crafting effective commit messages, refer to the How to write a git commit message guide. We prefer the commit message to be included in the PR body itself rather than a link to an external website (ie: Jira ticket).

- [ ] ♽ Before submitting a PR, run make test lint to avoid unnecessary CI processing. For an even more efficient workflow, consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the root of this repository.

- [ ] ✨ We use linters to maintain clean and consistent code. Please ensure you've run make lint before submitting a PR. Some linters offer a --fix mode, which can be executed with the command make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) tools are installed first).

- [ ] 📖 If you're introducing a user-facing feature or changing existing behavior, please ensure it's properly documented.

- [ ] 🧪 While 100% coverage isn't a requirement, we encourage unit tests for any code changes where possible.

- [ ] 🎁 If feasible, please check if an end-to-end test can be added. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for more details.

- [ ] 🔎 If there's any flakiness in the CI tests, don't necessarily ignore it. It's better to address the issue before merging, or provide a valid reason to bypass it if fixing isn't possible (e.g., token rate limitations).

chmouel commented on go.mod:
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift-pipelines/pipelines-as-code
 
-go 1.20
+go 1.21
```
can we hold on the go update?

chmouel commented on pkg/sort/runtime_sort.go:
```diff
@@ -80,7 +79,7 @@ func isLess(i, j reflect.Value) (bool, error) {
 	case reflect.Array, reflect.Slice:
 		// note: the length of i and j may be different
-		for idx := 0; idx < integer.IntMin(i.Len(), j.Len()); idx++ {
+		for idx := 0; idx < min(i.Len(), j.Len()); idx++ {
```
i think we can still use integer.Intmin to not require a go version bump,

chmouel commented:
I think the k8s update is unreleated but we may as well since i'll be doing a go mod update soon!

thanks!

savitaashture commented on go.mod:
```diff
@@ -40,7 +40,7 @@ require (
 	k8s.io/apimachinery v0.29.2
 	k8s.io/client-go v1.5.2
-	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
+	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
```
when we bump k8s again this get overrides 
so do we need to handle this again ??

sm43 replied:
removed those changes

sm43 committed:
Update sort package to latest & adds nil check for PRun before sorting

we use sort code from k8s repo, this updates the code to the latest master
but keeps using integer.IntMin to continue using golang 1.20
&
check pipelinrun is not nil and its name is not empty string
then only add for sorting
